src/riak_kv_index_hashtree.erl

```diff
@@ -296,11 +302,12 @@ handle_call({delete, Items}, _From, State) ->
 handle_call(get_trees, _From, #state{trees=Trees}=State) ->
     {reply, Trees, State};
 
-handle_call({update_tree, Id}, From, State) ->
+handle_call({update_tree, Id, Callback}, From, State) ->
```

Comment:
I think the guard needs to go on the API call.  In this particular case, there is a default/catch-all handle_call, which just returns ok.  But in this case I think we want the (API) call to fail if the type params don't match.

(I verified that if there were not a catch-all for handle_call, and we put a guard on the handle_call({update_re, ...), then the gen_server would crash, which is also probably what we don't want.)